code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -65,6 +72,13 @@
 	if(energy != oldenergy)
 		use_power(2500)
 
+/obj/machinery/chem_dispenser/emag_act(mob/user)
+	if(emagged || !emagged_reagents)
+		user << "<span class='warning'>\The [src] has no functional safeties to emag.</span>"
+		return
+	user << "<span class='notice'>You short out \the [src]'s safeties.</span>"
+	dispensable_reagents |= emagged_reagents//add the emagged reagents to the dispensable ones
```

Comment:
if somebody could help me with this:

my check on line 76 is supposed to return the code if the machine is already emagged or if it has a null emagged_reagents list. apparently it doesn't work, as emagging the machine multiple times will throw the message on line 79 multiple times (however it won't add infinite buttons to the chem dispenser so that's nice).